Modules/xASL_wrp_BIDS2Legacy.m

```diff
@@ -143,8 +145,18 @@
 if ~isempty(ListASL4D)
     for iList=1:numel(ListASL4D)
         xASL_bids_parseM0(ListASL4D{iList});
-        [~, currentNifti] = fileparts(ListASL4D{iList});
-        fprintf('M0 parsed for subject %s image %s ...\n', SubjectID, currentNifti);
+        [currentFolder, currentNifti] = xASL_fileparts(ListASL4D{iList});
+        [~, currentFolder] = fileparts(fileparts(currentFolder));
+
+        % Get session
+        sessionIndex = regexp(currentFolder, '_\d+');
```

Comment:
Better look for '_\d+$'

Reply:
This is outdated, because xASL_wrp_BIDS2Legacy is not there anymore (deleted in this issue).
BTW: I see now that we should then also delete `xASL_ut_function_xASL_wrp_BIDS2Legacy`?

Reply:
not delete. but correctly rename.

Reply:
`xASL_wrp_BIDS2Legacy` loaded the bids configuration, check the existence of xASLDerivatives, and started 

`xASL_bids_BIDS2Legacy_ParseModality`
M0 parsing
and 
`xASL_bids_BIDS2Legacy_CleanUp`

The loading is gone to `xASL_init_BIDS2Legacy`, and the rest all moved to `xASL_module_BIDS2Legacy`.

So we could change `xASL_ut_function_xASL_wrp_BIDS2Legacy` to 
`xASL_ut_function_xASL_module_BIDS2Legacy`, but this is tricky because these modules work with the status/lock GoNoGo system; and it won't have `xASL_init_BIDS2Legacy`.

So I don't see how to easily unit test, because this is not a "unit" anymore :)

Reply:
This function tests if BIDS2Legacy was correctly executed and generated the correct import by running ExploreASL(path). So it wasn't testing any subpart of the module, it tested the module in its entirety. You simply need to rename the function correctly and it will work. Just rename it to `xASL_ut_function_xASL_module_BIDS2Legacy` without even changing a line of code inside. 

If this test function doesn't work, it is a sign that BIDS2Legacy doesn't work.

Even if you keep the original name, the whole test works. So the test is still running, useful, and correct, it just should be renamed to avoid confusion.

Reply:
Done, sorry for my ignorance of our beautiful unit tests :)